translib/transformer/xfmr_intf.go

```diff
@@ -3144,3 +3212,326 @@ func retrievePortChannelAssociatedWithIntf(inParams *XfmrParams, ifName *string)
 	}
 	return nil, err
 }
+
+var YangToDb_intf_type_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	res_map := make(map[string]string)
+
+	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
+	intfType, _, err := getIntfTypeByName(ifName)
+	if intfType == IntfTypeUnset || err != nil {
+		log.Info("DbToYang_intf_type_xfmr - Invalid interface type IntfTypeUnset")
+		return res_map, errors.New("Invalid interface type IntfTypeUnset")
+	}
+
+	err = errors.New("Invalid interface type received")
```

Comment:
1) Avoid creating error object upfront
2) Use specific error type
3) I think "unsupported interface type" is sounds better as the value could be one of the valid enums defined by standard openconfig yang